### PR TITLE
Don't output example/exemplar files via CLI

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Add methods for track config
- Remove test and ignore regexp
- Exclude example/exemplar files from CLI filepath
- Fix bug

### CODE REVIEW

File: app/models/git/exercise.rb [hunk 1]
```diff
@@ -129,9 +139,7 @@ def tooling_files
     # This includes meta files
     memoize
     def tooling_filepaths
-      filepaths.select do |filepath| # rubocop:disable Style/InverseMethods
-        !filepath.match?(track.ignore_regexp)
-      end
+      filepaths
```
Comment (ErikSchierboom) [inline app/models/git/exercise.rb:142]: There's no special processing yet, but I still felt it was nice to have the method.

File: app/commands/submission/analysis/init.rb [hunk 2]
```diff
@@ -39,11 +39,6 @@ def exercise_filepaths
       def exercise_repo
         Git::Exercise.for_solution(solution)
       end
-
-      memoize
-      def test_regexp
-        solution.track.test_regexp
-      end
```
Comment (ErikSchierboom) [inline app/commands/submission/analysis/init.rb:46]: This wasn't even used

File: app/commands/submission/representation/init.rb [hunk 3]
```diff
@@ -36,11 +36,6 @@ def exercise_filepaths
       def exercise_repo
         Git::Exercise.for_solution(solution)
       end
-
-      memoize
-      def test_regexp
-        solution.track.test_regexp
-      end
```
Comment (ErikSchierboom) [inline app/commands/submission/representation/init.rb:43]: This wasn't even used

File: app/commands/submission/test_run/init.rb [hunk 4]
```diff
@@ -44,11 +44,6 @@ def exercise_filepaths
       def exercise_repo
         Git::Exercise.for_solution(solution)
       end
-
-      memoize
-      def test_regexp
-        solution.track.test_regexp
-      end
```
Comment (ErikSchierboom) [inline app/commands/submission/test_run/init.rb:51]: This wasn't even used

File: test/serializers/serialize_solution_for_cli_test.rb [hunk 5]
```diff
@@ -33,44 +33,6 @@ class SerializeSolutionForCLITest < ActiveSupport::TestCase
     assert_equal expected, SerializeSolutionForCLI.(solution, solution.user)
   end
 
-  # TODO: I think we don't need this test any more as it was
-  # only checking the exercise files, which are now checked elsewhere.
-  # But I'm leaving it here for now in case I was wrong.
```
Comment (ErikSchierboom) [inline test/serializers/serialize_solution_for_cli_test.rb:38]: I _think_ this can be removed now

File: app/models/git/track.rb [hunk 6]
```diff
@@ -19,14 +19,29 @@ def initialize(git_sha = "HEAD", repo_url: nil, repo: nil)
...
+    end
+
+    memoize
+    def tags
+      config[:tags].to_a
+    end
+
+    memoize
+    def active?
+      config[:active].present?
```
Comment (iHiD) [inline app/models/git/track.rb]: I have a small preference to `!!config[:active]` here. I wasn't immediately sure whether this handles `false` correctly (which it does). Also happy to stick with `present?` if you (or Rubocop) prefer it.

**Review by iHiD — APPROVED** [review]

File: app/models/git/track.rb (hunk 6 above)
Comment (ErikSchierboom) [reply app/models/git/track.rb]: Sure.

**Pull request merged**